modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -455,6 +493,14 @@ else if (needBackIfRouting) {
             // Do insert.
             L moveUpRow = p.insert(pageId, page, pageAddr, io, idx, lvl);
 
+            CacheDataRowAdapter row0 = null;
+
+            if (p.row instanceof CacheDataRowAdapter)
+                row0 = (CacheDataRowAdapter) p.row;
+
+            if (reconciliationCtx != null && row0 != null && reconciliationCtx.sizeReconciliationState(row0.cacheId()) == IN_PROGRESS)
```

Comment:
I think the Insert can only be done once. Otherwise the tests would fail.
But it is possible to Replace the key with the same key. This case is handled to avoid counting the same key twice.